e2e/dractions/actions.go

```diff
@@ -23,14 +23,18 @@ const (
 // Determine PVC label selector
 // Determine KubeObjectProtection requirements if Imperative (?)
 // Create DRPC, in desired namespace
+// nolint:funlen
 func EnableProtection(w workloads.Workload, d deployers.Deployer) error {
 	util.Ctx.Log.Info("enter EnableProtection " + w.GetName() + "/" + d.GetName())
 
 	name := GetCombinedName(d, w)
 	namespace := name
-	// if isAppSet {
-	// 	namespace = util.ArgocdNamespace
-	// }
+
+	_, isAppSet := d.(*deployers.ApplicationSet)
```

Comment:
(later) We need to resolve namespace detrmination and abstraction, potentially instead of inspecting this each time. I see this namespace as the deployer namespace, and for AppSet deployer it should return the namespace that it was configured with (from global configuration). For Subscriptions it can return the workload namespace. But the key si to abstract this out of actions.

Reply:
I moved the namespace decision code out to a function.